beacon-chain/core/helpers/legacy.go

```diff
@@ -0,0 +1,32 @@
+package helpers
+
+import (
+	"github.com/prysmaticlabs/prysm/v5/beacon-chain/state"
+	"github.com/prysmaticlabs/prysm/v5/math"
+	ethpb "github.com/prysmaticlabs/prysm/v5/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v5/runtime/version"
+)
+
+// IsLegacyDepositProcessPeriod determines if the current state should use the legacy deposit process.
+func IsLegacyDepositProcessPeriod(beaconState state.BeaconState, canonicalEth1Data *ethpb.Eth1Data) bool {
+	// Before the Electra upgrade, always use the legacy deposit process.
+	if beaconState.Version() < version.Electra {
+		return true
+	}
+
+	// Should not happen, but if the canonicalEth1Data is nil
+	// default to legacy
```

Comment:
is this fine?

Reply:
This should be impossible, I don't think this is the right approach